redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,637 @@
+                                {
+                                    const bool* functionalPtr =
+                                        std::get_if<bool>(&propertyMap.second);
+                                    if (functionalPtr == nullptr)
+                                    {
+                                        messages::internalError(asyncResp->res);
+                                        break;
+                                    }
+                                    state = *functionalPtr;
+                                }
+                            }
+                        }
+                    }
+                    asyncResp->res.jsonValue["@odata.type"] =
+                        "#License.v1_0_0.License";
+                    asyncResp->res.jsonValue["@odata.id"] =
+                        "/redfish/v1/LicenseService/Licenses/" + licenseEntryID;
+                    asyncResp->res.jsonValue["Id"] = licenseEntryID;
+                    asyncResp->res.jsonValue["SerialNumber"] = *serialNumPtr;
+                    asyncResp->res.jsonValue["Name"] = *licenseNamePtr;
+                    asyncResp->res.jsonValue["ExpirationDate"] =
+                        redfish::time_utils::getDateTimeUint(expirationTime);
+                    translateLicenseTypeDbusToRedfish(asyncResp,
+                                                      *licenseTypePtr);
+                    translateAuthorizationTypeDbusToRedfish(
+                        asyncResp, *authorizationTypePtr);
+                    asyncResp->res.jsonValue["MaxAuthorizedDevices"] =
+                        *deviceNumPtr;
+
+                    if (available)
+                    {
+                        asyncResp->res.jsonValue["Status"]["Health"] = "OK";
+                        if (state)
+                        {
+                            asyncResp->res.jsonValue["Status"]["State"] =
+                                "Enabled";
+                        }
+                        else
+                        {
+                            asyncResp->res.jsonValue["Status"]["State"] =
+                                "Disabled";
+                        }
+                    }
+                    else
+                    {
+                        asyncResp->res.jsonValue["Status"]["Health"] =
+                            "Critical";
+                        asyncResp->res.jsonValue["Status"]["State"] =
+                            "UnavailableOffline";
+                    }
+                    return;
+                }
+            }
+            BMCWEB_LOG_WARNING("LicenseEntry not found");
+            messages::resourceNotFound(asyncResp->res, "LicenseEntry",
+                                       licenseEntryID);
+        });
+}
+
+inline void requestRoutesLicenseEntry(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/LicenseService/Licenses/<str>/")
+        .privileges({{"Login"}})
```

Comment:
License is in https://redfish.dmtf.org/registries/Redfish_1.6.0_PrivilegeRegistry.json

Can't we use the standard privilege model ? getLicense